harness/determined/cli/workspace.py

```diff
@@ -111,39 +112,34 @@ def list_workspaces(args: Namespace) -> None:
 @authentication.required
 def list_workspace_projects(args: Namespace) -> None:
     sess = cli.setup_session(args)
-    w = workspace_by_name(sess, args.workspace_name)
-    orderArg = bindings.v1OrderBy[args.order_by.upper()]
-    sortArg = bindings.v1GetWorkspaceProjectsRequestSortBy[args.sort_by.upper()]
-    internal_offset = args.offset if ("offset" in args and args.offset) else 0
-    limit = args.limit if "limit" in args else 200
-    all_projects: List[bindings.v1Project] = []
-
-    while True:
-        projects = bindings.get_GetWorkspaceProjects(
-            sess,
-            id=w.id,
-            limit=limit,
-            offset=internal_offset,
-            orderBy=orderArg,
-            sortBy=sortArg,
-        ).projects
-        all_projects += projects
-        internal_offset += len(projects)
-        if ("offset" in args and args.offset) or len(projects) < limit:
-            break
+    all_projects = workspace.Workspace(
+        session=sess, workspace_name=args.workspace_name
+    ).list_projects()
+
+    sort_key = args.sort_by.lower()
+    sort_order = args.order_by.lower()
+    offset = args.offset or 0
+    limit = args.limit or 200
+
+    # TODO: Remove typechecking suppression when mypy is upgraded to 1.4.0
+    all_projects.sort(
+        key=lambda p: getattr(p, sort_key),  # type: ignore
```

Comment:
what happens if `sort_key` is invalid/doesn't exist on project? should we define a default or throw up?

Reply:
* If `sort_key` is invalid:
  - `sort_key` must be one of `["id", "name"]` as enforced by the argument parser. I'm not categorically opposed to adding validation to the method, too, but if after hearing this you still want it then I'd like to talk about it more
* If `sort_key` doesn't exist on project
  - Could you please clarify this for me? I didn't catch your meaning.

Reply:
i missed that it was enforced by the argument parser. cool.